projects/storefrontlib/src/lib/store-finder/services/google-map-renderer.service.ts

```diff
@@ -134,10 +122,21 @@ export class GoogleMapRendererService {
   }
 
   /**
-   * Moves all the markers to the given map
-   * @param map {@link google.maps.Map} the map where all the markers will be moved. Pass null if you just want to erase markers
+   * Initialize and draw the map
+   * @param mapElement {@link HTMLElement} inside of which the map will be drawn
+   * @param locations array of locations to be displayed on the map
+   * @param selectMarkerHandler function to handle whenever a marker on a map is clicked
    */
-  private setMapOnAllMarkers(map: google.maps.Map): void {
-    this.markers.forEach(marker => marker.setMap(map));
+  private drawMap(
+    mapElement: HTMLElement,
+    locations: any[],
+    selectMarkerHandler: Function
+  ) {
+    this.initMap(mapElement, this.defineMapCenter(locations));
+    if (selectMarkerHandler) {
```

Comment:
I think this condition is redundant as the very same verification is done inside createMarkers function. And to my mind createMarkers function is the only place where it should be done.